src/Analysis/Ast/Impl/Diagnostics/DiagnosticsEntry.cs

```diff
@@ -51,6 +53,19 @@ public sealed class DiagnosticsEntry {
         /// </summary>
         public DiagnosticSource Source { get; }
 
+        public bool ShouldReport(IPythonModule module) {
+            // module should always be a user written python module
+            if (module.ModuleType != ModuleType.User || !(module is PythonModule pythonModule)) {
+                return false;
+            }
+
+            if (module.GetAst().HasComment(pythonModule, "noqa", SourceSpan.Start.Line)) {
```

Comment:
`module.Analysis.Ast`. No need to cast to `PythonModule`

Reply:
Removed the cast.

Reply:
I need the cast for PythonModule.GetLine. It doesn't make sense to put that in IPythonModule because some PythonModules are not user code so getting a line from them is impossible.

Reply:
You can simply return an empty line for all others.

Reply:
I've gone ahead and put them in IPythonModule